### ISSUE TITLE
proper HTTP server example

### ISSUE THREAD

**Issue by weissi** [apple/swift-nio-examples#7]
we should have a short HTTP server example that does

- SSL
- HTTP/2
- possibly websockets

and it should live in this repository so we can depend on the other NIO parts. This should supersede `NIOHTTP1Server` as the example we point people to.

**Comment by natikgadzhi** [comment]
I was on the lookout for new adventures, but this one you claimed yourself back in 2019: https://github.com/apple/swift-nio-examples/pull/30 — time to close this.

**Comment by weissi** [comment]
@FranzBusch & @ktoso Could you comment here? Isn't a base-HTTP server a thing the SSWG wanted to look into?

**Comment by FranzBusch** [comment]
Yes the SSWG is investigating a common base HTTP-server. Progress is a bit slow at the moment but it still is on the roadmap.

**Comment by natikgadzhi** [comment]
As in a separate package that implements a base HTTP server based on NIO?
